ui/components/component-library/modal-header/deprecated/modal-header.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import classnames from 'classnames';
+import { HeaderBase, Text, ButtonIcon, ButtonIconSize, IconName } from '../..';
+import {
+  TextVariant,
+  TextAlign,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { ModalHeaderProps } from '.';
+
+/**
+ * @deprecated This version of `ModalHeader` is deprecated. Please use the version from the component-library in ui/components/component-library/modal-header/modal-header.tsx
+ * See PR https://github.com/MetaMask/metamask-extension/pull/22207 for details.
+ * @param options0
+ * @param options0.children
+ * @param options0.className
+ * @param options0.startAccessory
+ * @param options0.endAccessory
+ * @param options0.onClose
+ * @param options0.closeButtonProps
+ * @param options0.onBack
+ * @param options0.backButtonProps
```

Comment:
When I `command + S` this is automatically added and without it I get a linting error. So I think this is fine as this file will be removed in the near future

![Screenshot 2023-12-10 at 5 37 20 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/1e3c7163-7fe3-4f36-ad51-660fb8f2df89)